packages/node/src/integrations/tracing/fastify.ts

```diff
@@ -45,6 +58,20 @@ export function setupFastifyErrorHandler(fastify: Fastify): void {
         captureException(error);
       });
 
+      // registering `onRequest` hook here instead of using Otel `onRequest` callback b/c `onRequest` hook
+      // is ironically called in the fastify `preHandler` hook which is called later in the lifecycle:
+      // https://fastify.dev/docs/latest/Reference/Lifecycle/
+      fastify.addHook('onRequest', async (request, _reply) => {
+        const reqWithRouteInfo = request as FastifyRequestRouteInfo;
+
+        // Taken from Otel Fastify instrumentation:
+        // https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-fastify/src/instrumentation.ts#L94-L96
+        const routeName = reqWithRouteInfo.routeOptions?.url || reqWithRouteInfo.routerPath;
+        const method = reqWithRouteInfo.routeOptions?.method || 'GET';
+
+        getIsolationScope().setTransactionName(`${method} ${routeName}`);
```

Comment:
I am honestly wondering if we are grabbing the right isolation scope inside these hooks. Gut feeling wise it could I'd say no 🤔

Reply:
we should verify this for sure - there _may_ be a race condition if this hook is executed before the http.server span is created, that is correct 😬

Reply:
I tested this locally and the isolation scope already seems to be the forked one (i.e. not the default isolation scope). Added a check with a debug log nevertheless to ensure we don't pollute the default isolation scope.